cpp/src/arrow/flight/server.cc

```diff
@@ -102,6 +102,10 @@ class SerializationTraits<IpcPayload> {
 
     int64_t body_size = 0;
     for (const auto& buffer : msg.body_buffers) {
+      // Buffer may be null when the row length is zero, or when all
```

Comment:
It could be added in RecordBatchSerializer I believe, if you assume each column corresponds to exactly two buffers. By the time it gets to the IpcPayload serializer, that information is lost (unless you cast the metadata buffer back to the original type). The same comment is present in `writer.cc:176`.